### ISSUE TITLE
Wear OS

### ISSUE THREAD

**Issue by kj7rrv** [florisboard/florisboard#466]
There are no good open source keyboards for Wear OS (I'm not even sure there are any at all.) Would it be feasible to port FlorisBoard to Wear OS? (This would probably need to wait for glide typing, as that is basically necessary for typing on a circle 3cm in diameter.)

**Comment by patrickgold** [comment]
You can type on this small thingies? I would absolutely not be able to do that lol. Looking at existing Wear OS keyboards FlorisBoard's design and some functionalities could be copied to a WearOS edition, but this requires a full separation of the core and platform specific code to keep overlapping code as small as possible. I will add this one on my maybe todo list and see where I go with this.

**Comment by kj7rrv** [comment]
It's a little hard, but I do okay.
I did install the phone version on my watch just to see what it did, and the settings screen is actually usable and the keyboard renders okay but does not show a text box.
I will add screenshots to the original post.

**Comment by kj7rrv** [comment]
I was able to get the smartbar visible by changing to extra short, but still no textbox.

**Comment by kj7rrv** [comment]
Predictions do not work.
also, the Play Store just freezes on a black screen after tapping enter in FlorisBoard. I don't know what other apps do because the play store is the only app i have with a text box.

**Comment by kj7rrv** [comment]
Actually predictions do work! I just replied to a text using it. Google Play just doesn't like FlorisBoard on Wear OS. I just replied to a text using it.

**Comment by kj7rrv** [comment]
(Still no text box)

**Comment by kj7rrv** [comment]
T9 support would also really benefit Wear OS.

**Comment by patrickgold** [comment]
Well considering that FlorisBoard was designed for a medium sized phone screen it doesn't even looks that bad on a smartwatch... But yeah WearOS has its own rules for literally everything and thus it surprises me that it works in the state you got it working. for a good experience I'd have to write some WearOS specific code to fix crashes and bugs there. Like I said, maybe I'll do it but don't really know when exactly I will do this.

**Comment by kj7rrv** [comment]
Yeah, it looked pretty good to me too. At some point I might try making some changes to see if I can improve it a bit.On Mar 17, 2021 14:40, Patrick Goldinger ***@***.***> wrote:
Well considering that FlorisBoard was designed for a medium sized phone screen it doesn't even looks that bad on a smartwatch... But yeah WearOS has its own rules for literally everything and thus it surprises me that it works in the state you got it working. for a good experience I'd have to write some WearOS specific code to fix crashes and bugs there. Like I said, maybe I'll do it but don't really know when exactly I will do this.

**Comment by kj7rrv** [comment]
If it matters, those screenshots were from the unmodified 0.3.10beta01 APK installed via ADB from my phone
